src/python/pants/core_tasks/login.py

```diff
@@ -0,0 +1,39 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pants.auth.basic_auth import BasicAuth
+from pants.base.exceptions import TaskError
+from pants.task.task import Task
+
+
+class Login(Task):
```

Comment:
Having a task for this seems like overkill. Could this just be a memoized method instead?

Reply:
How would a user trigger this method?

Reply:
I realized why this is a synchronous task in my second comment, but forgot to remove this. This task is intended to be used directly... but clearly the `.netrc` usecase doesn't justify a manual login step, so I'm guessing you guys have some more advanced usage in mind.